src/runtime_src/core/pcie/driver/linux/xocl/subdev/xmc.c

```diff
@@ -733,7 +734,10 @@ static void xmc_sensor(struct platform_device *pdev, enum data_kind kind,
 			READ_SENSOR(xmc, XMC_VCCRAM_REG, val, val_kind);
 			break;
 		case XMC_POWER_WARN:
-			READ_SENSOR(xmc, XMC_POWER_WARN_REG, val, val_kind);
+			safe_read32(xmc, XMC_POWER_WARN_REG, val);
```

Comment:
read32 without val_kind means read offset+0 (SENSOR_MAX). Are you sure that is the value we are reading? I am confuzed.

Reply:
that one doesn't have max/min/ins defined. only offset 0 is available and it is the one they want